integrationTests/mock/accountsMap.go

```diff
@@ -12,10 +12,10 @@ type accountsMap struct {
 	accounts    map[string]*api.AccountResponse
 }
 
-func newAccountsMap() *accountsMap {
+func newAccountsMap(accounts map[string]*api.AccountResponse, contracts map[string]*Contract) *accountsMap {
```

Comment:
I would revert this as to not expose the maps without the mutex. Otherwise we might get into concurrent map access that will panic the program.
Instead, pass the accountsMap as a dependency to the contract notifier